content/api/webhooks.apib

```diff
@@ -4,54 +4,81 @@ description: Manage webhooks, which enable you to receive push updates of the ra
-
-## Webhooks Object Properties
-
-| Property          | Type   | Description | Required | Notes |
-|-------------------|--------|-------------|----------|-------|
-| name              | string | User-friendly name for webhook | yes | example: `Example webhook` |
-| target            | string | URL of the target to which to POST event batches | yes |  When a webhook is created or updated with a change to this property, a test POST request is sent to the given URL. The target URL must accept the connection and respond with HTTP 200; otherwise, your request to the Webhook API will fail with HTTP 400, and the requested change will not be applied.<br />example: `http://client.example.com/example-webhook` |
-| events            | array  | Array of event types this webhook will receive | yes | Use the Webhooks Events endpoint to list the available event types.<br />example: `["delivery", "injection", "open", "click"]`|
-| active            | boolean | Describes status of the webhook | no | Defaults to true, when false the target will no longer receive batches |
-| custom_headers    | JSON   | Object of custom headers to be used during POST requests to target | no | example: `{"x-api-key": "abcd"}` |
-| auth_type         | string | Type of authentication to be used during POST requests to target | no | examples: `none`, `basic`, `oauth2` |
-| auth_request_details | JSON | Object containing details needed to request authorization credentials, as necessary | no | example: `{ "url": "https://oauth.myurl.com/tokens", "body": { "client_id": "<oauth client id>", "client_secret": "<oauth client secret>" }}`|
-| auth_credentials         | JSON | Object containing credentials needed to make authorized POST requests to target | no | examples: `{ "access_token": "<oauth token>", expires_in: 3600 }`, `{ "username": "basicauthuser", "password": "mypassword" }` |
-| auth_token        | string | Authentication token to present in the `X-MessageSystems-Webhook-Token` header of POST requests to target | no | Use this token in your target application to confirm that data is coming from the Webhooks API. <br />example: `5ebe2294ecd0e0f08eab7690d2a6ee69`<br /><br /><span class="label label-danger"><strong>Deprecated</strong></span> in favor of the `auth_type` field. |
 
-__**The SparkPost webhooks API uses MaxMind software [MaxMind License](https://www.maxmind.com/download/geoip/database/LICENSE.txt)**__
+### Authentication
+
+You can use authentication to protect your webhook consumer and verify that only SparkPost is sending you data. We support both OAuth 2 and Basic authentication. When using OAuth 2, you need to set `auth_request_details`. The `auth_credentials` will be set by automatically from this. If you use Basic authentication, you need to set `auth_credentials` with a valid username and password.
```

Comment:
### Authentication

You can use authentication to protect your webhook consumer and verify that only SparkPost is sending you data.
We support both OAuth 2 and Basic authentication.

#### Basic Auth
For Basic authentication, set `auth_type` to `basic` and `auth_credentials` to your username and password.

    ```
    {
        "auth_type": "basic",
        "auth_credentials": {
            "username": "basicauthuser",
            "password": "mypassword"
    }
    ```

#### Oauth 2.0
Set `auth_type` to `oauth2` and `auth_request_details` as required.
`auth_credentials` will be automatically generated from the request details.

    ```
    {
        "auth_type": "basic",
        "auth_request_details": {
            "url": "http://client.example.com/tokens",
            "body": {
                "client_id": "CLIENT123",
                "client_secret": "9sdfj791d2bsbf",
                "grant_type": "client_credentials"
            }
        }
    }
    ```

-----------
Sample code might be unnecessary, but this should be outlined either here or in the Object